### ISSUE TITLE
[Bug] Keops kernels have no gradients when inputs are not contiguous

### ISSUE THREAD

**Issue by KeAWang** [cornellius-gp/gpytorch#1543]
# 🐛 Bug

## To reproduce

** Code snippet to reproduce **
```python
import torch
import gpytorch


make_contiguous = False  # change to `True` to get correct gradients

device = "cuda"
x = torch.randn(3, 1000, device=device)
x = x.transpose(0, 1).view(1000, 3)  # make non-contiguous
if make_contiguous:
    x = x.contiguous()
print(f"x is contiguous: {x.is_contiguous()}")
y = x.sum(-1) 

kernel = gpytorch.kernels.keops.RBFKernel().to(device)
kernel.train()
result = (kernel(x) @ y).sum()
result.backward()
print(f"raw_lengthscale.grad: {kernel.raw_lengthscale.grad}")
```

** Stack trace/error message **
```
x is contiguous: False
raw_lengthscale.grad: None
```

## Expected Behavior
Should print a non-zero gradient:
```
raw_lengthscale.grad: tensor([[-8274.9004]], device='cuda:0')
```
which you can reproduce by setting `make_contiguous = True`

## System information

**Please complete the following information:**
- GPyTorch version: '1.3.0', but also reproduces on `master`
- PyTorch version: '1.8.0'

## Some extra context
This is an issue because if `train_x` is ever not contiguous, we get a silent failure in the gradients (they all become None). A hotfix would just be to call `.contiguous` everywhere within the keops kernels, though I think this is an issue on the keops side.

**Comment by gpleiss** [comment]
@KeAWang could you put up a PR with the hot fix? That seems like a reasonable solution for now.

**Comment by gpleiss** [comment]
I think this issue was closed by #1544?
